src/main/java/reynev/kafkautils/kafka/message/MessageReader.java

```diff
@@ -37,14 +37,13 @@ public MessageReader(@Autowired Consumer<String, String> kafkaConsumer) {
     Iterable<ConsumerRecord<String, String>> readTopRecordsFromTopic(String topic, int amount){
         moveConsumerOffsetOnPartitions(topic, amount);
         ConsumerRecords<String, String> records = readMessages();
-        LimitedSortedSet<ConsumerRecord<String, String>> topRecords = getLatestMessages(amount, records);
-        return topRecords;
+        return getLatestMessages(amount, records);
     }
 
     private LimitedSortedSet<ConsumerRecord<String, String>> getLatestMessages(int amount, ConsumerRecords<String, String> records) {
         LimitedSortedSet<ConsumerRecord<String, String>> topRecords =
                 new LimitedSortedSet(new ConsumerRecordTimeComparator(), amount);
```

Comment:
Avoid using raw types. With raw types compilator cannot check whether collection/generics are used correctly. When type in diamond is added after new operator, then code does not compile anymore. It turns out that Comparator also uses ConsumerRecord with raw type.
It can be fixed by adding proper generics in Comparator.

Another option is to use Java8 style Comparator: new LimitedSortedSet<>(Comparator.comparingLong(ConsumerRecord::timestamp), amount);